bluepyopt/ephys/protocols.py

```diff
@@ -284,23 +281,22 @@ def _reduce_method(meth):
                                        sim=sim)
         return responses
 
-    def instantiate(self, sim=None, icell=None, LFPyCell=None):
+    def instantiate(self, sim=None, cell_model=None):
         """Instantiate"""
 
         for stimulus in self.stimuli:
-            if LFPyCell is not None:
-                stimulus.instantiate(sim=sim, icell=icell, LFPyCell=LFPyCell)
+            if isinstance(stimulus, LFPySquarePulse):
```

Comment:
@DrTaDa this is not flexible enough. In `multimodalfitting` we define a whole set of custom stimuli that also have LFP capabilities and would fail this. See [here](https://github.com/alejoe91/multimodalfitting/blob/master/multimodalfitting/ecode/ecodestimuli.py)

Reply:
I changed it in my last commit. If the stimulus classes in `multimodalfitting` inherit from the new `LFPStimulus` base class instead of `Stimulus`, it should work. (you would also have to change the parameters of the `instantiate` method of those classes)